readthedocs/search/faceted_search.py

```diff
@@ -92,24 +92,97 @@ class ProjectSearchBase(RTDFacetedSearch):
 class PageSearchBase(RTDFacetedSearch):
     facets = {
         'project': TermsFacet(field='project'),
-        'version': TermsFacet(field='version')
+        'version': TermsFacet(field='version'),
+        'role_name': NestedFacet(
+            'domains',
+            TermsFacet(field='domains.role_name')
+        ),
     }
     doc_types = [PageDocument]
     index = PageDocument._doc_type.index
-    fields = ['title^10', 'headers^5', 'content']
+
+    outer_fields = ['title']
+    section_fields = ['sections.title', 'sections.content']
+    domain_fields = [
```

Comment:
these 3 variables could perhaps have underscores, since they are our own variables, and not meaningful to the FacetedSearch class.

Reply:
That sounds more pythonic.
I have updated the PR.